### PR TITLE
Setup an integration test module to verify the behavior of code generators

### PR DESCRIPTION
**Pull request by vRallev**

### CODE REVIEW

File: integration-tests/code-generator/src/main/java/com/squareup/anvil/test/TestCodeGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        sequenceOf(
+          createGeneratedFile(
+            codeGenDir = codeGenDir,
+            packageName = generatedPackage,
+            fileName = "GeneratedClass",
+            content = generatedClass
+          )
+        )
+      }
```
Comment (vRallev) [inline integration-tests/code-generator/src/main/java/com/squareup/anvil/test/TestCodeGenerator.kt:48]: I'll extend this list later. It allows me to test various edge cases.

File: integration-tests/code-generator-tests/src/test/java/com/squareup/anvil/test/Utils.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.squareup.anvil.test
+
+import kotlin.reflect.KClass
+
+internal infix fun KClass<*>.extends(other: KClass<*>): Boolean =
+  other.java.isAssignableFrom(this.java)
```
Comment (christiandeange) [inline integration-tests/code-generator-tests/src/test/java/com/squareup/anvil/test/Utils.kt:6]: brb copying this into all of my projects because I can never remember the correct operand order for `isAssignableFrom`

File: settings.gradle [hunk 3]
```diff
@@ -2,6 +2,8 @@ include ':annotations'
 include ':compiler'
 include ':compiler-api'
 include ':compiler-utils'
+include ':integration-tests:code-generator'
+include ':integration-tests:code-generator-tests'
```
Comment (christiandeange) [inline settings.gradle:6]: Any particular reason the contents of this module couldn't just go into `/integration-tests/code-generator/src/test/java`?

**Review by christiandeange — APPROVED** [review]

File: settings.gradle (hunk 3 above)
Comment (vRallev) [reply settings.gradle:6]: Yes, because the main source set would need to be added to the Kotlin compiler plugin classpath. There's no separate classpath for the main and test source set.

File: integration-tests/code-generator-tests/src/test/java/com/squareup/anvil/test/Utils.kt (hunk 2 above)
Comment (vRallev) [reply integration-tests/code-generator-tests/src/test/java/com/squareup/anvil/test/Utils.kt:6]: Same :-D
